File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcCartesianTransformationOperator/IfcCartesianTransformationOperator2D/IfcCartesianTransformationOperator2D.cs
public class IfcCartesianTransformationOperator2D : IfcCartesianTransformationOperator
{

    // === EXTENDED BY ===
    // IfcCartesianTransformationOperator2DnonUniform

    // === WHERE CLAUSES ===
    // Axis1Is2D : NOT(EXISTS(SELF\IfcCartesianTransformationOperator.Axis1)) OR (SELF\IfcCartesianTransformationOperator.Axis1.Dim = 2)
    // Axis2Is2D : NOT(EXISTS(SELF\IfcCartesianTransformationOperator.Axis2)) OR (SELF\IfcCartesianTransformationOperator.Axis2.Dim = 2)
    // DimEqual2 : SELF\IfcCartesianTransformationOperator.Dim = 2

    // === DERIVE CLAUSES ===
    // U : LIST [2:2] OF IfcDirection := IfcBaseAxis(2,SELF\IfcCartesianTransformationOperator.Axis1, SELF\IfcCartesianTransformationOperator.Axis2,?)
}

File: output/ifc4x3_dev_73740fe4/entity/IfcCartesianTransformationOperator2D.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


import java.util.List;

public class IfcCartesianTransformationOperator2D extends IfcCartesianTransformationOperator {

    // INVERSE attributes:
    // U : LIST [2:2] OF IfcDirection := IfcBaseAxis(2,SELF\IfcCartesianTransformationOperator.Axis1,
    // SELF\IfcCartesianTransformationOperator.Axis2,?);
    // INVERSE attributes:
    // Axis1Is2D : NOT(EXISTS(SELF\IfcCartesianTransformationOperator.Axis1)) OR
    // (SELF\IfcCartesianTransformationOperator.Axis1.Dim = 2);
    // Axis2Is2D : NOT(EXISTS(SELF\IfcCartesianTransformationOperator.Axis2)) OR
    // (SELF\IfcCartesianTransformationOperator.Axis2.Dim = 2);
    // DimEqual2 : SELF\IfcCartesianTransformationOperator.Dim = 2;

    // SUPERTYPE OF:
    // - IfcCartesianTransformationOperator2DnonUniform SUBTYPE OF IfcCartesianTransformationOperator

    // WHERE constraints:
    // Axis1Is2D : NOT(EXISTS(SELF\IfcCartesianTransformationOperator.Axis1)) OR
    // (SELF\IfcCartesianTransformationOperator.Axis1.Dim = 2);
    // Axis2Is2D : NOT(EXISTS(SELF\IfcCartesianTransformationOperator.Axis2)) OR
    // (SELF\IfcCartesianTransformationOperator.Axis2.Dim = 2);
    // DimEqual2 : SELF\IfcCartesianTransformationOperator.Dim = 2;
}
